python/pfs/utils/optimizeBlackDots.py

```diff
@@ -0,0 +1,501 @@
+        """Fit polynomial fit to the measurments
+
+        Parameters
+        ----------
+        n_obs: `int`
+            Number of observations
+        measurments: `np.array`
+            Measurments of the crossing experiment
+        isGood: `np.array`
+            Array contaning information if the fiber was seen
+            in this particular observation
+        poly_order: `int`
+            Which poly order to fit to the measured data
+
+        Returns
+        ----------
+        fit_to_single_point: `np.array`
+            Polynomials factors of the fit
+        residuals: `np.array`
+            Residual to the fit
+
+        Notes
+        ----------
+        Gets called by `predict_positions`
+
+        """
+        fit_for_single_fiber, residuals =\
+            np.polyfit(np.arange(0, n_obs)[isGood],
+                       measurments[isGood], poly_order, full=True)[0:2]
+
+        return fit_for_single_fiber, residuals
+
+    @staticmethod
+    def check(x_obs, y_obs, xd, yd, r):
+        """Check if the cobra is covered by the black dot
+
+        Parameters
+        ----------
+        x_obs: `np.array` of `float`, (N_fib, N_obs)
+            x coordinates of the observations [mm]
+        y_obs: `np.array` of `float``, (N_fib, N_obs)
+            y coordinate of the observations [mm]
+        xd: `np.array` of `float``, (N_fib,)
+            x coordinate of the dot [mm]
+        yd: `np.array` of `float`, (N_fib,)
+            y coordinate of the dot [mm]
+        r: `float`
+            Radius of dot [mm]
+
+        Returns
+        ----------
+        cover value: `bool`
+            Return True if the cobra is covered
+
+        Notes
+        ----------
+        Gets called by `total_penalty_for_single_dot`
+        """
+        x1 = (x_obs.T-xd).T
+        y1 = (y_obs.T-yd).T
+
+        return np.hypot(x1, y1) < r
+
+    def new_position_of_dots(self, aff_mat_11, aff_mat_21, aff_mat_12,
+                             aff_mat_22, aff_mat_13, aff_mat_23):
+        """Move the dots to the new positions via affine tranformation
+
+        Parameters
+        ----------
+        aff_mat_11: `float`
+            Value for affine matrix at position [1,1]
+        aff_mat_12: `float`
+            Value for affine matrix at position [2,1]
+        aff_mat_21: `float`
+            Value for affine matrix at position [1,2]
+        aff_mat_12: `float`
+            Value for affine matrix at position [2,2]
+        aff_mat_13: `float`
+            Value for affine matrix at position [1,3]
+        aff_mat_23
+            Value for affine matrix at position [2,3]
+
+        Returns
+        ----------
+        dots_new: `pd.dataframe`
+            Dataframe contaning final guesses for dots
+            The columns are ['spotId', 'x', 'y', 'r']
+        """
+        dots = self.dots
+        xd_original = dots['x'].values
+        yd_original = dots['y'].values
+        AffineTransform_instance = AffineTransform()
```

Comment:
Names starting with a capital letter are supposed to be classes. Why not just call it `transform`?